gripper_action_controller/include/gripper_action_controller/gripper_action_controller.hpp

```diff
@@ -0,0 +1,188 @@
+// Copyright 2014, SRI International
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// \author Sachin Chitta, Adolfo Rodriguez Tsouroukdissian
+
+#ifndef GRIPPER_ACTION_CONTROLLER__GRIPPER_ACTION_CONTROLLER_HPP_
+#define GRIPPER_ACTION_CONTROLLER__GRIPPER_ACTION_CONTROLLER_HPP_
+
+// C++ standard
+#include <cassert>
+#include <memory>
+#include <stdexcept>
+#include <string>
+// TODO(JafarAbdi): Remove experimental once the default standard is C++17
+#include "experimental/optional"
+
+// ROS
+#include "rclcpp/rclcpp.hpp"
+
+// ROS messages
+#include "control_msgs/action/gripper_command.hpp"
+
+// rclcpp_action
+#include "rclcpp_action/create_server.hpp"
+
+// ros_controls
+#include "controller_interface/controller_interface.hpp"
+#include "gripper_action_controller/visibility_control.hpp"
+#include "hardware_interface/loaned_command_interface.hpp"
+#include "hardware_interface/loaned_state_interface.hpp"
+#include "realtime_tools/realtime_buffer.h"
+#include "realtime_tools/realtime_server_goal_handle.h"
+
+// Project
+#include "gripper_action_controller/hardware_interface_adapter.hpp"
+
+namespace gripper_action_controller
+{
+
+/**
+ * \brief Controller for executing a gripper command action for simple
+ * single-dof grippers.
+ *
+ * \tparam HardwareInterface Controller hardware interface. Currently \p
+ * hardware_interface::PositionJointInterface and \p
+ * hardware_interface::EffortJointInterface are supported out-of-the-box.
```

Comment:
Is this still actual?

```suggestion
 * \tparam HardwareInterface Controller hardware interface. Currently \p
 * hardware_interface::PositionJointInterface and \p
 * hardware_interface::EffortJointInterface are supported out-of-the-box.
```

Reply:
I think we should replace them with `hardware_interface::HW_IF_POSITION` and  `hardware_interface::HW_IF_EFFORT`, right .?

Reply:
yes please